php/libraries/NDB_Page.class.inc

```diff
@@ -620,7 +620,11 @@ class NDB_Page implements RequestHandlerInterface
     ) : ResponseInterface {
         $user = $request->getAttribute("user");
         if ($this->_hasAccess() !== true) {
-            throw new PermissionDenied();
+            return (new \LORIS\Middleware\PageDecorationMiddleware($user))
+                ->process(
+                    $request,
+                    new \LORIS\Http\StringStream("Permission Denied")
```

Comment:
Feels like the error message could be more indicative. Maybe something like:
"You do not have permission to view candidates that do not belong to your site" or something like that?

Reply:
But this middleware is used for every page, regardless of the reason.